toucan_connectors/google_sheets/google_sheets_connector.py

```diff
@@ -88,17 +100,32 @@ class GoogleSheetsConnector(ToucanConnector, data_source_model=GoogleSheetsDataS
     _auth_flow = "managed_oauth2"
     _managed_oauth_service_id = "google-sheets"
     _oauth_trigger = "retrieve_token"
-    _retrieve_token: Callable[[str, str], str] = PrivateAttr()
+    _retrieve_token: Callable[[str, str], str] | None = PrivateAttr()
 
     auth_id: PlainJsonSecretStr = None
 
-    def __init__(self, retrieve_token: Callable[[str, str], str], *args, **kwargs):
+    def __init__(self, retrieve_token: Callable[[str, str], str] | None = None, *args, **kwargs):
         super().__init__(**kwargs)
         self._retrieve_token = retrieve_token  # Could be async
```

Comment:
I don't think `__init__` should be overloaded on models.. could you please remove it ?